clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1200,12 +1200,22 @@ static std::string eraseAnonNamespace(std::string S) {
 }
 
 // Creates a mangled kernel name for given kernel name type
-static std::string constructKernelName(QualType KernelNameType,
-                                       MangleContext &MC) {
+std::string constructKernelName(QualType KernelNameType, MangleContext &MC,
+                                MangleNumberingContext *MNCtx) {
   SmallString<256> Result;
   llvm::raw_svector_ostream Out(Result);
 
   MC.mangleTypeName(KernelNameType, Out);
+
+  if (MNCtx) {
+    auto *RecordDecl = KernelNameType->getAsCXXRecordDecl();
```

Comment:
```suggestion
    auto *LabdaCallOperator = KernelNameType->getAsCXXRecordDecl()->getLambdaCallOperator();
```
... and using it in two function calls below. Maybe combining those into one if you see fit.

Reply:
@AGindinson actually makes sense to me